File: service-course/application/adapters/repositories/models/video.go
package models

import "time"

type VideoResolution struct {
	Resolution string `bson:"resolution"`
	URL        string `bson:"url"`
	Status     string `bson:"status"`
}

type VideoModel struct {
	Id          string            `bson:"_id"`
	Type        string            `bson:"type"`
	TmpUrl      string            `bson:"tmpurl"`
	Url         string            `bson:"url"`
	Status      string            `bson:"status"`
	Duration    float32           `bson:"duration"`
	Size        int64             `bson:"size"`
	Resolutions []VideoResolution `bson:"resolutions"`
	UserId      string            `bson:"userId"`
	CreatedAt   time.Time         `bson:"createdAt"`
	UpdatedAt   time.Time         `bson:"updatedAt"`
}

File: front/types/video.ts
export type Resolution = {
  resolution: string;
  complete_resolution: string;
  url: string;
};

export type Size = {
  low: number;
  high: number;
  unsigned: boolean;
};

export type Video = {
  resolutions: Resolution[];
  id: string;
  type: string;
  status: string;
  size: Size;
  createdAt: string;
  updatedAt: string;
  url: string;
};
